src/metabase/driver/sql_jdbc/sync/describe_table.clj

```diff
@@ -395,7 +396,7 @@
   (with-open [conn (jdbc/get-connection spec)]
     (let [table-identifier-info [(:schema table) (:name table)]
           table-fields          (describe-table-fields driver conn table nil)
-          json-fields           (filter #(= (:semantic-type %) :type/SerializedJSON) table-fields)]
+          json-fields           (filter #(isa? (:base-type %) :type/JSON) table-fields)]
```

Comment:
This is tested for both [postgres](https://github.com/metabase/metabase/blob/2774c1cdb47520dcceefaee8d9c0f6b5375e25ca/test/metabase/driver/postgres_test.clj#L490) and [mysql](https://github.com/metabase/metabase/blob/2774c1cdb47520dcceefaee8d9c0f6b5375e25ca/test/metabase/driver/mysql_test.clj#L429)